SpatialGDK/Source/SpatialGDK/Public/Interop/InitialOnlyFilter.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "SpatialConstants.h"
+#include "SpatialView/ComponentData.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogInitialOnlyFilter, Log, All);
+
+class USpatialNetDriver;
+
+namespace SpatialGDK
+{
+class InitialOnlyFilter
+{
+public:
+	InitialOnlyFilter(USpatialNetDriver* InNetDriver);
```

Comment:
Does this need to be the net driver? It generally makes things really hard to test and it looks like in every case what it actually wants is the view coordinator. Likewise the view coordinator is preferable to the worker connection as it adds ownership safety and uses the newer types.

Reply:
The view coordinator will cover all of the cases except for setting the delegate to receive the queries, but for that we could perhaps use `SpatialOSDispatcherInterface`?

Reply:
Can pass in the coordinator directly. All GDK components use the receiver for entity queries currently. I'm all for cleaning it up, but I think that can be done fully in a separate PR.

Reply:
Provided something passes this the view delta it can read get hold of queries that it cares about (and if it has a coordinator it can even do that itself in an advance call).

Reply:
I think the ticket move command responses out of catchalls is currently being worked on.

Reply:
> I think the ticket move command responses out of catchalls is currently being worked on.

Cool, if that lands before this PR, I'll update, otherwise I'll make sure it gets done in that PR.

Reply:
That PR is likely just moving each thing one by one though. We could just cut out the middle man and go straight to it and then we don't introduce another class with a dependency on a monolithic global object.